nectar/src/command/trade/event_loop.rs

```diff
@@ -87,97 +87,95 @@ impl EventLoop {
                     }
                 },
                 event = self.swarm.next().fuse() => {
-                    self.handle_network_event(
-                            event,
-                        ).await?;
+                    let _ = self.handle_network_event(
+                        event,
+                    ).await.map_err(|err| tracing::error!("Network event handling failed: {:#}", err));
                 },
-                rate_update = self.rate_update_receiver.next().fuse() => {
-                    self.handle_rate_update(rate_update.unwrap());
+                new_rate = self.rate_update_receiver.next().fuse() => {
+                    if let Some(new_rate) = new_rate {
+                        match new_rate {
+                            Ok(new_rate) => {
+                                let _ = self.handle_rate_update(new_rate)
+                                    .map_err(|err| tracing::error!("Rate update handling failed: {:#}", err));
+                            }
+                            Err(err) => tracing::error!("Rate retrieval failed: {:#}", err),
+                        }
+                    }
```

Comment:
You can combine this to:

```suggestion
if let Some(Ok(new_rate)) = new_rate {
	let _ = self.handle_rate_update(new_rate).map_err(|err| tracing::error!("Rate update handling failed: {:#}", err));
}
```

If you want to keep the error logging, use `match` and match on `Some(Err(e))`.

Reply:
and introducing a `map_err` that only has a side-effect with the data being dropped? 😱